File: 2 тиждень/caesar/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

int main(int a, string b[])
{
    int key = 0;
    if (a!=2)
    {
        printf("Usage: ./caesar key\n");
        return 0;
    }
    for(int i=0, n=strlen(b[1]);i<n;i++)
    {
        if(b[1][i]<'0'||b[1][i]>'9')
        {
            printf("Usage: ./caesar key\n");
            return 1;
        }
    }
    key =  atoi ( b[1]);
    if (key>=26)
    {
        key%=26;
    }
    string vvid = get_string("plaintext:  ");
    printf("ciphertext: ");
    for (int i=0, n=strlen(vvid);i<n;i++)
    {
        int chislo = vvid[i];
        if(chislo>=97&&chislo<=122)
        {
            chislo+=key;
            if (chislo>122)
            {
                int ost = chislo-122;
                chislo = 97+ost-1;
                printf("%c",chislo);
            }
            else 
            {
                printf("%c",chislo);
            }
        }
        else if(chislo>=65&&chislo<=90)
        {
            chislo+=key;
            if(chislo>90)
            {
                int ost = chislo-90;
                chislo=65+ost-1;
                printf("%c",chislo);
            }
            else
            {
                printf("%c",chislo);
            }
        }
        else 
        {
            printf("%c", vvid[i]);
        }
    }
    printf("\n");
}

File: 6 тиждень/caesar/caesar.py
import sys
from cs50 import get_string
from sys import argv
if len(argv) != 2:
    print("Usage: python caesar.py k ")
    sys.exit(1)
kod = int(argv[1])
kod %= 26
if kod < 0:
    print("Usage: python caesar.py k ")
    sys.exit(1)
vvid = get_string("plaintext:  ")
print("ciphertext: ", end="")
for c in vvid:
    symbol = ord(c)
    if symbol >= 97 and symbol <= 122:
        symbol += kod
        if symbol > 122:
            symbol -= 26
        print(chr(symbol), end="")
    elif symbol >= 65 and symbol <= 90:
        symbol += kod
        if symbol > 90:
            symbol -= 26
        print(chr(symbol), end="")
    else:
        print(chr(symbol), end="")
print()
sys.exit(0)





